pkgs/by-name/st/starpu/package.nix

```
@@ -0,0 +1,118 @@
+{
+  # derivation dependencies
+  lib,
+  fetchurl,
+  stdenv,
+
+  # starpu dependencies
+  hwloc,
+  libuuid,
+  libX11,
+  fftw,
+  fftwFloat, # Same than previous but with float precision
+  pkg-config,
+  libtool,
+  autoconf,
+  automake,
+  simgrid,
+  mpi,
+  cudaPackages,
+
+  # Options
+  enableSimgrid ? false,
+  enableMPI ? false,
+  enableCUDA ? false,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "starpu";
+  version = "1.4.7";
+
+  inherit enableSimgrid;
+  inherit enableMPI;
+  inherit enableCUDA;
+
+  src = fetchurl {
+    url = "https://files.inria.fr/starpu/starpu-${finalAttrs.version}/starpu-${finalAttrs.version}.tar.gz";
+    hash = "sha256-HrPfVRCJFT/m4LFyrZURhDS0qB6p6qWiw4cl0NtTsT4=";
+  };
+
+  # Runtime build dependencies
+  nativeBuildInputs =
+    [
+      pkg-config
+      hwloc
+    ]
+    ++ lib.optional finalAttrs.enableSimgrid simgrid
```

```suggestion
    ++ lib.optional stdenv.hostPlatform.isDarwin "--enable-maxcpus=4"
    ++ lib.optional finalAttrs.enableSimgrid simgrid
```
This configure option just sets the default, and the auto-detect fails on Darwin